lib/extractor/index.ts

```diff
@@ -1,13 +1,58 @@
-import { ImageType } from "../types";
+import {
+  getLayersFromPackages,
+  getPackagesFromRunInstructions,
+} from "../dockerfile/instruction-parser";
+import { AutoDetectedUserInstructions, ImageType } from "../types";
 import * as dockerExtractor from "./docker-archive";
 import * as ociExtractor from "./oci-archive";
 import {
+  DockerArchiveManifest,
   ExtractAction,
   ExtractedLayers,
+  ExtractedLayersAndManifest,
   ExtractionResult,
+  Extractor,
   FileContent,
+  ImageConfig,
+  OciArchiveManifest,
 } from "./types";
 
+export class InvalidArchiveError extends Error {}
+class ArchiveExtractor {
+  private extractor: Extractor;
+  private fileSystemPath: string;
+  private extractActions: ExtractAction[];
+
+  constructor(extractor: Extractor, path: string, actions: ExtractAction[]) {
+    this.fileSystemPath = path;
+    this.extractActions = actions;
+    this.extractor = extractor;
+  }
+
+  public getExtractor(): Extractor {
+    return this.extractor;
+  }
+
+  public async getLayersAndManifest(): Promise<ExtractedLayersAndManifest> {
+    return await this.extractor?.extractArchive(
```

Comment:
In which scenario the extractor is undefined?

Reply:
None, it was "leftover" from a previous version where it could have been, but I changed it as it seemed silly 😄